### PR TITLE
Fixes storage client resource name in list page

### PR DESCRIPTION
**Pull request by bipuladh**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=2262975

### CODE REVIEW

File: packages/shared/src/types/storage.ts [hunk 1]
```diff
@@ -177,6 +177,9 @@ type StorageConsumerStatus = {
 type ClientStatus = {
   platformVersion: string;
   operatorVersion: string;
+  clusterId: string;
+  clusterName: string;
```
Comment (GowthamShanmugam) [inline packages/shared/src/types/storage.ts:181]: clusterName is not used anywhere

File: packages/odf/components/storage-consumers/client-list.tsx [hunk 2]
```diff
@@ -83,7 +82,7 @@ const ClientsList: React.FC<ClientListProps> = (props) => {
     () => [
       {
         title: t('Name'),
-        sort: 'metadata.name',
+        sort: 'status.client.name',
```
Comment (GowthamShanmugam) [inline packages/odf/components/storage-consumers/client-list.tsx:85]: dont we need an option symbol here like status?.client?.name? Does this won't break when the client list is empty?

Comment (bipuladh) [reply packages/odf/components/storage-consumers/client-list.tsx:85]: This is used inside of lodash get.

File: packages/shared/src/types/storage.ts (hunk 1 above)
Comment (bipuladh) [reply packages/shared/src/types/storage.ts:181]: However they have updated the type of ClientStatus so even if it's not used now the type definition can be useful for later.

Comment (GowthamShanmugam) [reply packages/shared/src/types/storage.ts:181]: ok

**Review by GowthamShanmugam — APPROVED** [review]

**Comment by alfonsomthd** [comment]
LGTM

**Pull request merged**